tree/ntuple/v7/inc/ROOT/RDaos.hxx

```diff
@@ -111,20 +111,21 @@ public:
       FetchUpdateArgs() = default;
       FetchUpdateArgs(const FetchUpdateArgs&) = delete;
       FetchUpdateArgs(FetchUpdateArgs&& fua);
-      FetchUpdateArgs(DistributionKey_t &d, AttributeKey_t &a, std::vector<d_iov_t> &v, bool is_async = false);
+      FetchUpdateArgs(const DistributionKey_t &d, const std::vector<AttributeKey_t> &&as,
+                      const std::vector<d_iov_t> &&vs, bool is_async = false);
```

Comment:
In the constructor initializer list at `RDaos.cxx:77`, it seems that the intented behavior is for `fAkeys` and `fIovs` to adopt the storage passed vectors.  However, note that the vector is still copied due to the use of a const rvalue reference, i.e. `const std::vector<...> &&`.  In this specific case, the `const` qualifier should be removed.

Or even better: given that we can ensure that the `FetchUpdateArgs` lifetime at least covers that of the sync/async operation, we could probably make both `fAkeys` and `fIovs` a `std::span<...>`.  This construction would then also take two `std::span<...>`-s.  This should be tested though; if it works, the comment below has to be also updated.